event-log/src/main/scala/io/renku/eventlog/subscriptions/EventsDispatcher.scala

```diff
@@ -109,12 +109,13 @@ class EventsDispatcher(
     } yield ()
   }
 
-  private def loggingError[O](retry: => IO[O]): PartialFunction[Throwable, IO[O]] = { case NonFatal(exception) =>
-    for {
-      _      <- logger.error(exception)("Finding events to dispatch failed")
-      _      <- timer sleep onErrorSleep
-      result <- retry
-    } yield result
+  private def loggingErrorAndRetry[O](retry: () => IO[O]): PartialFunction[Throwable, IO[O]] = {
+    case NonFatal(exception) =>
+      for {
+        _      <- logger.error(exception)("Finding events to dispatch failed")
+        _      <- timer sleep onErrorSleep
+        result <- retry() recoverWith loggingErrorAndRetry(retry)
```

Comment:
So is it ok if we retry indefinitely?

Reply:
I believe so as for instance postgres could be down for some time but eventually should come back